cmd/kube-proxy/app/server.go

```diff
@@ -159,9 +171,14 @@ func (o *Options) AddFlags(fs *pflag.FlagSet) {
 	fs.Float32Var(&o.config.ClientConnection.QPS, "kube-api-qps", o.config.ClientConnection.QPS, "QPS to use while talking with kubernetes apiserver")
 
 	fs.StringVar(&o.hostnameOverride, "hostname-override", o.hostnameOverride, "If non-empty, will be used as the name of the Node that kube-proxy is running on. If unset, the node name is assumed to be the same as the node's hostname.")
-	fs.Var(&utilflag.IPVar{Val: &o.config.BindAddress}, "bind-address", "Overrides kube-proxy's idea of what its node's primary IP is. Note that the name is a historical artifact, and kube-proxy does not actually bind any sockets to this IP. This parameter is ignored if a config file is specified by --config.")
-	fs.Var(&utilflag.IPPortVar{Val: &o.config.HealthzBindAddress}, "healthz-bind-address", "The IP address and port for the health check server to serve on, defaulting to \"0.0.0.0:10256\" (if --bind-address is unset or IPv4), or \"[::]:10256\" (if --bind-address is IPv6). Set empty to disable. This parameter is ignored if a config file is specified by --config.")
-	fs.Var(&utilflag.IPPortVar{Val: &o.config.MetricsBindAddress}, "metrics-bind-address", "The IP address and port for the metrics server to serve on, defaulting to \"127.0.0.1:10249\" (if --bind-address is unset or IPv4), or \"[::1]:10249\" (if --bind-address is IPv6). (Set to \"0.0.0.0:10249\" / \"[::]:10249\" to bind on all interfaces.) Set empty to disable. This parameter is ignored if a config file is specified by --config.")
+	fs.StringArrayVar(&o.config.NodeIPOverride, "node-ip-override", o.config.NodeIPOverride, "Overrides kube-proxy's idea of what its node's primary IP is. This parameter is ignored if a config file is specified by --config.")
+	fs.Var(&utilflag.IPVar{Val: &o.bindAddress}, "bind-address", "Overrides kube-proxy's idea of what its node's primary IP is. Note that the name is a historical artifact, and kube-proxy does not actually bind any sockets to this IP. This parameter is ignored if a config file is specified by --config.")
+	fs.Var(&utilflag.IPPortVar{Val: &o.healthzAddress}, "healthz-bind-address", "The IP address and port for the health check server to serve on, defaulting to \"0.0.0.0:10256\" (if --bind-address is unset or IPv4), or \"[::]:10256\" (if --bind-address is IPv6). Set empty to disable. This parameter is ignored if a config file is specified by --config.")
+	fs.StringArrayVar(&o.config.HealthzBindAddresses, "healthz-bind-addresses", o.config.HealthzBindAddresses, "The list is a list of CIDR ranges that contains a valid node IP on which health check server will be served on. Set empty to disable. This parameter is ignored if a config file is specified by --config.")
```

Comment:
Hm... I had thought we discussed this but I guess it didn't make it into the KEP update...

We absolutely do not want "This parameter is ignored if a config file is specified" for v1alpha2. That's really really confusing, and we would have fixed it, except that it would be a config API break at this point. But v1alpha2 gives us the opportunity to do that.

So we should do one of:
1. If you use a `v1alpha2` config file, then you can't pass any other command-line flags (ie, kube-proxy will exit with an error). Thus, there is no need to add new command-line flags for the new v1alpha2 options.
2. Like 1, except a few command-line flags are allowed for inherently-per-node config options (in particular, `--node-ip-override` and `--hostname-override`. Maybe nothing else.) In that case the command-line flags override the config file, not vice versa.
3. Like 2, except _all_ flags are allowed, and there are new flags for new options, and all command-line flags override a `v1alpha2` config file (but _not_ a `v1alpha1` config file).

I think I was assuming 2. Not sure what other people think. (@aojea?)

FWIW there are no command-line flags for nftables config, and nobody has complained (or even _commented_ on that fact). I think very few people deploy kube-proxy "by hand", and all of the k8s deploy/install tools (kubeadm, etc) write out config files because it's cleaner...

Reply:
I like the second option, kubelet is also doing the same thing - https://github.com/kubernetes/enhancements/pull/2158#discussion_r616788544
https://github.com/kubernetes/kubernetes/blob/24267f6aa78d0b5b957dc618c7c4dac5dfa21fa3/cmd/kubelet/app/options/options.go#L335-L363

Reply:
Should we then also mark all the existing flags except logging/instance specific deprecated?

Reply:
we can't do that yet; kubelet has a v1 config format

Reply:
understood!
If we don't deprecate the existing flags yet and don't add any new flags then we are good, right?
(and only allow select flags [node-ip-override, host-name-override] more precedence over config)

Reply:
> FWIW there are no command-line flags for nftables config, and nobody has complained (or even _commented_ on that fact).

Antonio eventually complained/commented; [the GCE setup scripts](https://github.com/kubernetes/kubernetes/blob/37ca0377d542d9eb5b6a7ebac2093b8ec6fc9942/cluster/gce/gci/configure-helper.sh#L1781) (which are used for k8s CI among other things) use command-line flags rather than config files. (Because they predate the addition of `--config`...)

Reply:
I think option 2 makes sense, and I think it is important to behave similarly to other components (kubelet and kube-proxy are like cousins).  To recap, tell me if I understood correctly:

* Most options would be configurable EITHER by flags or a config file, but not both (applying to the whole set, not individually)
* A small set of options would be configurable by flags AND a config file (who wins if both are specified?)
* The only flag which is not also a config file option is `--config`